### ISSUE TITLE
Opening some community feeds displays no posts and incorrect top menu buttons.

### ISSUE THREAD

**Issue by roskelld** [Memmy-App/memmy#598]
**Describe the bug**
When clicking on a community link from the feed menu it loads the community page but states that there are no posts and the top three right menu buttons stay configured to the feed menu and do not update to the community menu (missing the block community option). 
This only happens with some communities and I'm not sure what is the distinction. I found that it happened on linuxmemes hosted on Lemmyworld and acab hosted on lemmygrad.ml 

**To Reproduce**
1. Go to feed
2. Click on community link on feed post
3. See top right menu buttons and missing post feed. 

**Screenshots**

![image](https://github.com/Memmy-App/memmy/assets/18073522/7de99517-6496-4512-814e-e1d93e34fa37)

![image](https://github.com/Memmy-App/memmy/assets/18073522/dfd193fa-349d-4f8c-8853-d153737eb82d)

**Smartphone (please complete the following information):**
 - Device: iPhone 7 Plus
 - OS: iOS 15

**Comment by eoddc5** [comment]
what version of the app are you on? I cannot replicate this issue

**Comment by roskelld** [comment]
Testflight latest 0.1.0.14

I had to go through quite a few communities before finding broken ones. It was a case of going down All and clicking on each community link until the bug appeared.

**Comment by gkasdorf** [comment]
If I had to take a guess without looking at code, I'd say the header probably is not changing because the re-render must reply on the posts state being changed perhaps? I noticed that it doesn't actually change this just based off of switching to a community (watch how the header doesn't update until the posts load).

As for why no posts are showing up, I'm going to assume it's because you are using 1 hour top sort, which is (in my opinion) partly a Lemmy issue but also something we could handle better.

I'm not sure if Lemmy plans to actually fix this, so let me look at their issues and see if it is being discussed. We can implement some sort of change here to have it update to top six hours whenever it runs out of content.
